drivers/flash/flash_partition.c

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright (c) 2021 Laczen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/flash_partition.h>
+
+/* Helper functions */
+#define OUTSIDE_RANGE(val, min, max) (val < min || val > max)
+
+static int read_arguments_check(const struct flp_info_cfg *cfg, off_t off,
+				size_t len)
+{
+
+	if (OUTSIDE_RANGE(off, 0, cfg->size) ||
```

Comment:
@nvlsianpu, thank you for taking the time to review.

The macro OUTSIDE_RANGE checks for both the min value and the max value, when off is negative this will return an error.